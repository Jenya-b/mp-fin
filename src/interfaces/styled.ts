export interface ITheme {
  colors: {
    backgroundBase: string;
    backgroundPrimary: string;
    backgroundSecondary: string;
    backgroundTertiary: string;
    backgroundTelegramPrimary: string;
    backgroundTelegramSecondary: string;
    textBase: string;
    textPrimary: string;
    textSecondary: string;
    textTertiary: string;
    textAttentionPrimary: string;
    textAttentionSecondary: string;
    borderPrimary: string;
    borderSecondary: string;
    borderTertiary: string;
  };
  sizes: {
    loginForm: {
      width: number;
    };
    inputForm: {
      height: number;
    };
    logoForm: {
      width: number;
    };
    primaryBtn: {
      height: number;
    };
    sidebar: {
      widthActive: number;
      widthHidden: number;
      logo: { height: number };
    };
    header: {
      height: number;
    };
    footer: {
      height: number;
    };
    notFound: {
      title: { fontSize: number };
    };
    errorFallback: {
      title: { fontSize: number };
    };
    settingInputFile: {
      width: number;
      height: number;
    };
    replenishmentBlock: {
      height: number;
    };
    adminForm: {
      maxWidth: number;
      width: number;
    };
    telegramIcon: {
      width: number;
      height: number;
    };
  };
  indents: {
    loginForm: {
      paddingTop: number;
      paddingRight: number;
      paddingBottom: number;
      paddingLeft: number;
    };
    inputList: {
      marginTop: number;
      rowGap: number;
    };
    main: {
      paddingLeft: number;
      paddingLeftHide: number;
    };
    adminMenu: {
      paddingTop: number;
      paddingRight: number;
      paddingBottom: number;
      paddingLeft: number;
    };
    adminList: {
      rowGap: number;
    };
    adminContainer: {
      marginTop: number;
    };
    adminForm: {
      columnGap: number;
    };
    adminSubtitle: {
      marginBottom: number;
    };
    adminSearchBlock: {
      marginTop: number;
    };
    queryTableRow: {
      paddingTop: number;
      paddingBottom: number;
    };
  };
  borders: {
    loginForm: { borderRadius: number };
    inputForm: { borderRadius: number };
    primaryBtn: { borderRadius: number };
    replenishment: { borderRadius: number };
  };
  transitions: {
    input: { time: number };
    button: { time: number };
  };
  order: {
    header: number;
    sidebar: number;
    settingInputFile: number;
    settingLogo: number;
  };
  cursor: string;
  media: {
    extraLarge: string;
    large: string;
    medium: string;
    small: string;
  };
}
