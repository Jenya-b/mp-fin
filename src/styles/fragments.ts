import { css } from 'styled-components';

export const flexCenterAll = css`
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const linkStyles = css`
  a {
    color: ${({ theme }) => theme.colors.formLink};

    :hover {
      color: ${({ theme }) => theme.colors.formLinkHover};
    }
  }
`;

export const ButtonMailStyles = css`
  border-radius: ${({ theme }) => theme.borders.primaryBtn.borderRadius}px;
  background: ${({ theme }) => theme.colors.btnbackground};
  transition: all ${({ theme }) => theme.transitions.button.time}s;

  :hover,
  :active {
    background: ${({ theme }) => theme.colors.btnHover};
  }
`;
