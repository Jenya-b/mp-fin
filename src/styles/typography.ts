import { css } from 'styled-components';

const fontsFamilyOpenSans = css`
  font-family: 'Open Sans', sans-serif;
  font-style: normal;
`;

const fontsFamilyInter = css`
  font-family: 'Inter', sans-serif;
  font-style: normal;
`;

export const fontStylesH1 = css`
  ${fontsFamilyOpenSans}
  font-weight: 600;
  font-size: 28px;
  line-height: 38px;
`;
export const fontStylesH2 = css`
  ${fontsFamilyOpenSans}
  font-weight: 600;
  font-size: 22px;
  line-height: 30px;
`;
export const fontStylesRegularBold = css`
  ${fontsFamilyOpenSans}
  font-weight: 600;
  font-size: 18px;
  line-height: 25px;
`;
export const fontStylesRegular = css`
  ${fontsFamilyOpenSans}
  font-weight: 400;
  font-size: 18px;
  line-height: 25px;
`;
export const fontStylesCaptionBig = css`
  ${fontsFamilyOpenSans}
  font-weight: 600;
  font-size: 16px;
  line-height: 22px;
`;
export const fontStylesCaption = css`
  ${fontsFamilyInter}
  font-weight: 400;
  font-size: 14px;
  line-height: 22px;
`;
export const fontStylesSmall = css`
  ${fontsFamilyInter}
  font-weight: 400;
  font-size: 12px;
  line-height: 22px;
`;
