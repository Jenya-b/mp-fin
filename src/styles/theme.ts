import { ITheme } from 'interfaces/styled';
import { Colors } from 'constants/colors';

export const baseTheme: ITheme = {
  colors: {
    backgroundBase: Colors.MainBlue,
    backgroundPrimary: Colors.MainWhite,
    backgroundSecondary: Colors.MainGreenSecondary,
    backgroundTertiary: Colors.MainGreenPrimary,
    backgroundTelegramPrimary: Colors.AdditionalBlue,
    backgroundTelegramSecondary: Colors.AdditionalBlue,
    textBase: Colors.MainBlack,
    textPrimary: Colors.MainGreySecondary,
    textSecondary: Colors.MainGreyLight,
    textTertiary: Colors.MainGreyPrimary,
    textAttentionPrimary: Colors.MainBluePrimary,
    textAttentionSecondary: Colors.AdditionalPinkSecondary,
    borderPrimary: Colors.MainGreyLight,
    borderSecondary: Colors.MainGreenSecondary,
    borderTertiary: Colors.MainGreyTertiary,
  },
  sizes: {
    loginForm: {
      width: 390,
    },
    inputForm: {
      height: 55,
    },
    logoForm: {
      width: 590,
    },
    primaryBtn: {
      height: 55,
    },
    sidebar: {
      widthActive: 290,
      widthHidden: 90,
      logo: { height: 80 },
    },
    header: {
      height: 80,
    },
    footer: {
      height: 60,
    },
    notFound: {
      title: { fontSize: 56 },
    },
    errorFallback: {
      title: { fontSize: 56 },
    },
    settingInputFile: {
      width: 178,
      height: 178,
    },
    replenishmentBlock: {
      height: 239,
    },
    adminForm: {
      maxWidth: 900,
      width: 100,
    },
    telegramIcon: {
      width: 25,
      height: 25,
    },
  },
  indents: {
    loginForm: {
      paddingTop: 39,
      paddingRight: 45,
      paddingBottom: 39,
      paddingLeft: 45,
    },
    inputList: {
      marginTop: 25,
      rowGap: 15,
    },
    main: {
      paddingLeft: 290,
      paddingLeftHide: 90,
    },
    adminMenu: {
      paddingTop: 40,
      paddingRight: 25,
      paddingBottom: 40,
      paddingLeft: 30,
    },
    adminList: {
      rowGap: 10,
    },
    adminContainer: {
      marginTop: 30,
    },
    adminForm: {
      columnGap: 10,
    },
    adminSubtitle: {
      marginBottom: 10,
    },
    adminSearchBlock: {
      marginTop: 30,
    },
    queryTableRow: {
      paddingTop: 7,
      paddingBottom: 7,
    },
  },
  borders: {
    loginForm: { borderRadius: 10 },
    inputForm: { borderRadius: 10 },
    primaryBtn: { borderRadius: 10 },
    replenishment: { borderRadius: 10 },
  },
  transitions: {
    input: { time: 0.3 },
    button: { time: 0.3 },
  },
  // z-index
  order: {
    sidebar: 50,
    settingInputFile: 10,
    settingLogo: 1,
  },
  cursor: 'pointer',
  media: {
    extraLarge: '(max-width: 1279px)',
    large: '(max-width: 1023px)',
    medium: '(max-width: 767px)',
    small: '(max-width: 479px)',
  },
};
