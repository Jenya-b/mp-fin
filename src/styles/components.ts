import styled from 'styled-components';
import { ButtonMailStyles } from 'styles/fragments';
import {
  fontStylesCaption,
  fontStylesH1,
  fontStylesH2,
  fontStylesRegularBold,
} from 'styles/typography';

export const Main = styled.main`
  padding: 80px 60px;
`;

export const MainTitle = styled.h1`
  ${fontStylesH1}
  color: ${({ theme }) => theme.colors.mainTitle};
`;

export const MainSubtitle = styled.h2`
  ${fontStylesH2}
  color: ${({ theme }) => theme.colors.mainTitle};
`;

export const Button = styled.button`
  color: ${({ theme }) => theme.colors.btn};
  ${ButtonMailStyles}
`;

export const PrimaryButton = styled(Button)`
  width: 100%;
  height: ${({ theme }) => theme.sizes.primaryBtn.height}px;
  ${fontStylesRegularBold}
`;

export const Input = styled.input`
  width: 100%;
  height: ${({ theme }) => theme.sizes.inputForm.height}px;
  padding-left: 20px;
  ${fontStylesCaption}
  color: ${({ theme }) => theme.colors.inputFormText};
  border-radius: ${({ theme }) => theme.borders.inputForm.borderRadius}px;
  border: 1px solid ${({ theme }) => theme.colors.inputFormBorder};
  transition: all ${({ theme }) => theme.transitions.input.time}s;

  ::placeholder {
    color: ${({ theme }) => theme.colors.inputFormPlaceholder};
  }

  :hover {
    border: 1px solid ${({ theme }) => theme.colors.inputFormHoverBord};
  }

  :focus {
    border: 1px solid ${({ theme }) => theme.colors.inputFormFocusBord};
  }
`;

export const PrimaryInput = styled(Input)`
  background: ${({ theme }) => theme.colors.inputFormBackground};
`;

export const SearchInput = styled(Input)`
  background: ${({ theme }) => theme.colors.inputFormBackground};
`;

export const Checkbox = styled.input``;
