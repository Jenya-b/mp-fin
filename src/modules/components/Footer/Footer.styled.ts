import styled from 'styled-components';
import { flexCenterAll } from 'styles/fragments';

export const StyledFooter = styled.footer`
  background: ${({ theme }) => theme.colors.footerBg};
  border: 1px solid ${({ theme }) => theme.colors.footerBorder};
  ${flexCenterAll}
`;

export const FooterInfo = styled.p``;
