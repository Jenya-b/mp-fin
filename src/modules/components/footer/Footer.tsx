import styled from 'styled-components';
import { flexCenterAll } from '../../../styles/fragments';

export const Footer = () => (
  <StyledFooter>
    <FooterInfo>© MpFin, 2022. Все права защищены</FooterInfo>
  </StyledFooter>
);

const StyledFooter = styled.footer`
  background: ${({ theme }) => theme.colors.footerBg};
  border: 1px solid ${({ theme }) => theme.colors.footerBorder};
  ${flexCenterAll}
`;

const FooterInfo = styled.p``;
