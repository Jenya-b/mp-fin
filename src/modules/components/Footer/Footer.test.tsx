import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { baseTheme } from 'styles/theme';
import { Footer } from './Footer';
import { StyledFooter, FooterInfo } from './Footer.styled';

describe('Snapshot Footer styled-components', () => {
  test('StyledFooter', () => {
    const tree = renderer.create(<StyledFooter theme={baseTheme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('FooterInfo', () => {
    const tree = renderer.create(<FooterInfo />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Footer component', () => {
  test('it renders', () => {
    render(
      <ThemeProvider theme={baseTheme}>
        <Footer />
      </ThemeProvider>
    );
    expect(screen.getByText('© MpFin, 2022. Все права защищены')).toBeInTheDocument();
  });
});
