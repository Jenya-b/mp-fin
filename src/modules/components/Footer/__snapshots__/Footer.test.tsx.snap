// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot styled-components FooterInfo 1`] = `
<p
  className="sc-gswNZR"
/>
`;

exports[`Snapshot styled-components StyledFooter 1`] = `
<footer
  className="sc-bcXHqe gdjWve"
/>
`;
